cloud-provider.sh

```diff
@@ -38,48 +39,65 @@ set_azure_config() {
   fi
 
   if [ -z "$az_location" ]; then
-    az_location=$(curl  -s -H Metadata:true "${AZURE_META_URL}/location?api-version=2017-08-01&format=text")
+    az_location=$(curl  -s -H Metadata:true "${AZURE_META_URL}/location?api-version=${AZURE_META_API_VERSION}&format=text")
   fi
 
-  vmcmd="vm" # default is "standard" and needs "vm" cmd
   if [ "$az_vm_type" = "vmss" ]; then
-    vmcmd="vmss" # scale set cmd
-  fi
+    # vmss
+    local az_vm_scale_set_name=$(curl  -s -H Metadata:true "${AZURE_META_URL}/vmScaleSetName?api-version=${AZURE_META_API_VERSION}&format=text")
+    local az_vm_instance_id=$(az vmss list-instances -g ${az_resources_group} --name ${az_vm_scale_set_name} --query "[?name=='${az_vm_name}'].instanceId" --output tsv)
+    local az_vm_nic=$(az vmss nic list -g ${az_resources_group} --vmss-name ${az_vm_scale_set_name} --output json --query [0].id)
```

Comment:
Is there a specific reason for `.id` and the post processing vs `query [0].name`?

Reply:
its possible that was added in later api versions and not inthe 2017 one this was built on.. im going to update it cause it's much cleaner.